health_vara.py

```diff
@@ -64,9 +70,24 @@ def search_mobile(cls, _name, clause):
         value = clause[2]
 
         res.append(('contact_mechanisms.type', '=', 'mobile'))
-        res.append(('contact_mechanisms.value', op, value))
+        res.append(('contact_mechanisms.rec_name', op, value))
```

Comment:
Makes "specific" mobile search work without spaces:
![image](https://github.com/vara-ai/health_vara/assets/1367955/5cb198c2-f44e-4549-ba7c-c0aa09aa2058)

Reply:
I am curious what `contact_mechanisms.value` holds. Is it also a string or is it a reference to another table?

Reply:
`value` is a string like `+20 101 111 1003`, on `contact_mechanisms` there's another field `value_compact` which is a string like `+201011111003`.

`rec_name` is a special thing that every tryton class implements and defines how global search works, how the value is displayed in the UI etc, for example patient.rec_name is `first-name + last-name` when displayed, and the search domain for rec_name does many SQL things.

For contact mechanism, the search by `rec_name` basically does an or statement between `value` and `value_compact` to see if a string matches either one, hence why this change works.

Reply:
Thanks for the explanation. :)